src/libs/ReportUtils.ts

```diff
@@ -1823,44 +1835,58 @@ function canEditMoneyRequest(reportAction: OnyxEntry<ReportAction>, fieldToEdit
     }
 
     const moneyRequestReport = getReport(String(moneyRequestReportID));
-    const isReportSettled = isSettled(moneyRequestReport?.reportID);
-    const isApproved = isReportApproved(moneyRequestReport);
-    const isAdmin = isExpenseReport(moneyRequestReport) && (getPolicy(moneyRequestReport?.policyID ?? '')?.role ?? '') === CONST.POLICY.ROLE.ADMIN;
+    const policy = getPolicy(moneyRequestReport?.policyID ?? '');
+    const isAdmin = isExpenseReport(moneyRequestReport) && policy.role === CONST.POLICY.ROLE.ADMIN;
+    const isManager = isExpenseReport(moneyRequestReport) && currentUserAccountID === moneyRequestReport?.managerID;
     const isRequestor = currentUserAccountID === reportAction?.actorAccountID;
 
-    if (isAdmin && !isRequestor && fieldToEdit === CONST.EDIT_REQUEST_FIELD.RECEIPT) {
-        return false;
+    // Admin & managers can always allow coding fields such as tag, category, billable, etc. As long as the report has a state higher than OPEN.
+    if ((isAdmin || isManager) && !isOpenReport(moneyRequestReport)) {
+        return true;
     }
 
-    if (isAdmin) {
-        return true;
+    // We allow editing of PROCESSING reports in Free policies, but not in paid policie (yet)
+    if (policy.type === CONST.POLICY.TYPE.FREE) {
+        return isRequestor && isProcessingReport(moneyRequestReport);
     }
 
-    return !isApproved && !isReportSettled && isRequestor;
+    return !isReportApproved(moneyRequestReport) && !isSettled(moneyRequestReport?.reportID) && isRequestor;
 }
 
-/**
- * Checks if the current user can edit the provided property of a money request
- *
- */
-function canEditFieldOfMoneyRequest(reportAction: OnyxEntry<ReportAction>, reportID: string, fieldToEdit: ValueOf<typeof CONST.EDIT_REQUEST_FIELD>): boolean {
+function canEditFieldOfMoneyRequest(reportAction: OnyxEntry<ReportAction>, transaction: Transaction, fieldToEdit: ValueOf<typeof CONST.EDIT_REQUEST_FIELD>): boolean {
     // A list of fields that cannot be edited by anyone, once a money request has been settled
-    const nonEditableFieldsWhenSettled: string[] = [
+    const restrictedFields: string[] = [
         CONST.EDIT_REQUEST_FIELD.AMOUNT,
         CONST.EDIT_REQUEST_FIELD.CURRENCY,
+        CONST.EDIT_REQUEST_FIELD.MERCHANT,
         CONST.EDIT_REQUEST_FIELD.DATE,
         CONST.EDIT_REQUEST_FIELD.RECEIPT,
         CONST.EDIT_REQUEST_FIELD.DISTANCE,
     ];
 
-    // Checks if this user has permissions to edit this money request
-    if (!canEditMoneyRequest(reportAction, fieldToEdit)) {
-        return false; // User doesn't have permission to edit
+    if (!canEditMoneyRequest(reportAction)) {
+        return false;
+    }
+
+    if (!restrictedFields.includes(fieldToEdit)) {
+        return true;
+    }
+
+    const reportID = reportAction?.originalMessage?.IOUReportID ?? 0;
```

Comment:
At the other two places, where we are using this `reportAction?.originalMessage?.IOUReportID` reportAction is narrowed to `CONST.REPORT.ACTIONS.TYPE.IOU`. If we narrow it down here as well typescript will stop complaining.
```suggestion
    const reportID = reportAction?.actionName === CONST.REPORT.ACTIONS.TYPE.IOU ? reportAction?.originalMessage?.IOUReportID : 0;
```

Reply:
Great catch, thanks! I'm going to cast instead because we know for sure the type of the action here